src/System.Management.Automation/namespaces/IDynamicPropertyProvider.cs

```diff
@@ -90,7 +92,7 @@ public interface IDynamicPropertyCmdletProvider : IPropertyCmdletProvider
         object? NewPropertyDynamicParameters(
             string path,
             string propertyName,
-            string propertyTypeName,
+            RegistryValueKind propertyType,
```

Comment:
We can not change public API, and it is an interface. So it was my wrong proposal.

Reply:
In this case the only option is to use ToString() on `RegistryValueKind` in cmdlet, but then so much converting back and forth to get registry value.

Reply:
Or we can create overloads for public api to not change existing one. I feel that option is too much work though.

Reply:
Better to close the PR.

Reply:
> Better to close the PR.

Had a feeling it would be too hard. Oh well, at least I tried and understand internals a lot more now with registry provider.